tools/common/CoreResolver.cs

```diff
@@ -102,17 +102,17 @@ public virtual AssemblyDefinition Load (string fileName)
 			return assembly;
 		}
 
-		protected AssemblyDefinition SearchDirectory (string name, string directory, string extension = ".dll")
+		protected AssemblyDefinition SearchDirectory (string name, string directory, string extension = ".dll", bool recursive = false)
 		{
-			var file = DirectoryGetFile (directory, name + extension);
+			var file = DirectoryGetFile (directory, name + extension, recursive);
 			if (file.Length > 0)
 				return Load (file);
 			return null;
 		}
 
-		static string DirectoryGetFile (string directory, string file)
+		static string DirectoryGetFile (string directory, string file, bool recursive)
 		{
-			var files = Directory.GetFiles (directory, file);
+			var files = Directory.GetFiles (directory, file, recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly);
 			if (files != null && files.Length > 0)
 				return files [0];
```

Comment:
Please throw an error or warning if searching recursively and more than one assembly with the same name is found. That makes it easier to diagnose any resulting problems.